### ISSUE TITLE
Expose supported upload file formats as configurations

### ISSUE THREAD

**Issue by DavidQuartz** [GeoNode/geonode#9591]
# GNIP <num> - Expose supported upload file formats as configurations

## Overview

Describe briefly and clearly what this proposal is meant for.
To start the testing of GeoPackage imports from the UI the .gpkg extension must be included in the supported extensions for the client.
This option is experimental and shouldn't land in master right away. Therefore, instead of creating a dedicated client branch, we want to take the opportunity to make the formats configurable, and let the client obtain the configurations from GeoNode.

The idea is make the client get [this configuration](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/routes/UploadDataset.jsx#L27-L76) from a new property `__GEONODE__CONFIG__.localSettings.geonodeSettings.upload.supportedDatasetsFileTypes` (notice the additional `File` infix, to clarify the purpose of this settings).

GeoNode will expose this configuration inside the goenode_config template, taking it from the settings.

I'm proposing a new `upload` configuration level where we can gather anything related to the upload configuration. For example, the `documentMaxUploadSize` could be moved here too.

This change will be merged only once the backend is ready to provide such a configuration.

### Proposed By
@giohappy 
<user and full name of the proponent>

### Assigned to Release

This proposal is for GeoNode <version>.

### State

* [x] Under Discussion
* [x] In Progress
* [x] Completed
* [ ] Rejected
* [ ] Deferred

### Motivation

Details about the motivations. Why people should accept this proposal. What are the benefits compared to the current situation.

## Proposal

Technical details for developers.

### Backwards Compatibility

Declare its Backwards Compatibility.

## Future evolution

Explain which could be future evolutions.

## Feedback

Update this section with relevant feedbacks, if any.

## Voting

Project Steering Committee:

* Alessio Fabiani:
* Francesco Bartoli:
* Giovanni Allegri:
* Simone Dalmasso:
* Toni Schoenbuchner:
* Florian Hoedt:

## Links

Remove unused links below.

* [Email Discussion]()
* [Pull Request]()
* [Mail Discussion]()
* [Linked Issue]()

**Comment by t-book** [comment]
Hi is this GNIP still active? I cannot find it here https://github.com/GeoNode/geonode/wiki/GeoNode-Improvement-Proposals
and if so it still needs voting ?
